src/versionUtils.ts

```diff
@@ -101,31 +105,46 @@ export async function pingGithubUrl(urlStr: string) : Promise<boolean> {
 }
 
 export async function getLatestCamelKVersion(): Promise<Errorable<string>> {
-	const latestURL = 'https://api.github.com/repos/apache/camel-k/releases/latest';
-	const downloadResult = await downloader.toTempFile(latestURL);
-	if (failed(downloadResult)) {
-		return { succeeded: false, error: [`Failed to establish Apache Camel K stable version: ${downloadResult.error[0]}`] };
-	}
-	const rawtext = fs.readFileSync(downloadResult.result, 'utf-8');
-	let latestJSON = JSON.parse(rawtext);
-	let tagName = latestJSON.tag_name;
-	if (tagName) {
-		return { succeeded: true, result: tagName };
+	if (latestVersionFromOnline) {
+		return { succeeded: true, result: latestVersionFromOnline };
+	} else {
+		const latestURL = 'https://api.github.com/repos/apache/camel-k/releases/latest';
+		const headers = [['If-Modified-Since', LAST_MODIFIED_DATE_OF_DEFAULT_VERSION]];
+		let githubToken = process.env.VSCODE_CAMELK_GITHUB_TOKEN;
+		if(githubToken) {
+			headers.push(['Authorization', `token ${githubToken}`]);
+		}
+		const res = await fetch(latestURL, { headers: headers });
+		if (res.status === 200) {
+			let latestJSON = await res.json();
+			let tagName = latestJSON.tag_name;
+			if (tagName) {
+				latestVersionFromOnline = tagName;
+				return { succeeded: true, result: tagName };
+			} else {
+				return { succeeded: false, error: [`Failed to retrieve latest Apache Camel K version tag from : ${latestURL}`] };
+			}
+		} else if (res.status === 304) {
+			latestVersionFromOnline = version;
+			return { succeeded: true, result: latestVersionFromOnline };
+		} else {
+			console.log(`error ${res.status} ${res.statusText}`);
+			return { succeeded: false, error: [`Failed to establish Apache Camel K stable version: ${res.status} ${res.statusText}`] };
+		}
 	}
-	return { succeeded: false, error: [`Failed to retrieve latest Apache Camel K version tag from : ${latestURL}`] };
 }
 
-function checkKamelCLIVersion() : Promise<string> {
-	return new Promise<string>( async (resolve, reject) => {
+function checkKamelCLIVersion(): Promise<string> {
+	return new Promise<string>(async (resolve, reject) => {
 		let kamelLocal = kamelCli.create();
```

Comment:
const instead of let

Reply:
updated as a it is a trivial change even if it wasn't part of modified code